### PR TITLE
`azurerm_network_watcher_flow_log` - support for `version`

### LINKED ISSUE #5403: Support for version 2 flow logs in azurerm_network_watcher_flow_log
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
When you enable flow logs on a NSG, the NSG supports two kinds of versions. Version 1 and version 2.

Extract from Azure Portal: "Version 1 logs ingress and egress IP traffic flows for both allowed and denied traffic. Version 2 provides additional throughput information (bytes and packets) per flow." 

Azurerm_network_watcher_flow_log therefore should have an option or switch to choose which version of flow logs are needed.

### New or Affected Resource(s)

* azurerm_network_watcher_flow_log

### Potential Terraform Configuration

```hcl
resource "azurerm_network_watcher_flow_log" "test" {
  network_watcher_name = "${azurerm_network_watcher.test.name}"
  resource_group_name  = "${azurerm_resource_group.test.name}"

  network_security_group_id = "${azurerm_network_security_group.test.id}"
  storage_account_id        = "${azurerm_storage_account.test.id}"
  enabled                   = true

  flow_log_version = v1/v2

  retention_policy {
    enabled = true
    days    = 7
  }

  traffic_analytics {
    enabled               = true
    workspace_id          = "${azurerm_log_analytics_workspace.test.workspace_id}"
    workspace_region      = "${azurerm_log_analytics_workspace.test.location}"
    workspace_resource_id = "${azurerm_log_analytics_workspace.test.id}"
  }
}
```

**flow_log_version = v1/v2 beeing a new option.**

### References

https://docs.microsoft.com/en-us/azure/network-watcher/network-watcher-nsg-flow-logging-overview
and
https://github.com/terraform-providers/terraform-provider-azurerm/pull/5059

* #0000

### PR DESCRIPTION
**Pull request by aqche**
Fixes #5403

Adds the `version` argument for the `azurerm_network_watcher_flow_log` resource.

### CODE REVIEW

File: azurerm/internal/services/network/resource_arm_network_watcher_flow_log.go [hunk 1]
```diff
@@ -253,6 +264,7 @@ func resourceArmNetworkWatcherFlowLogRead(d *schema.ResourceData, meta interface
 
 	if props := fli.FlowLogProperties; props != nil {
 		d.Set("enabled", props.Enabled)
+		d.Set("version", props.Format.Version)
```
Comment (katbyte) [inline azurerm/internal/services/network/resource_arm_network_watcher_flow_log.go]: We'll need to nil check `Format` here

File: azurerm/internal/services/network/resource_arm_network_watcher_flow_log.go [hunk 2]
```diff
@@ -145,6 +145,13 @@ func resourceArmNetworkWatcherFlowLog() *schema.Resource {
 					},
 				},
 			},
+
+			"version": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
```
Comment (katbyte) [inline azurerm/internal/services/network/resource_arm_network_watcher_flow_log.go]: Is it possible to remove the default and set this to computed instead?

File: website/docs/r/network_watcher_flow_log.html.markdown [hunk 3]
```diff
@@ -88,6 +88,8 @@ The following arguments are supported:
 
 * `traffic_analytics` - (Optional) A `traffic_analytics` block as documented below.
 
+* `version` - (Optional) The version (revision) of the flow log.
```
Comment (katbyte) [inline website/docs/r/network_watcher_flow_log.html.markdown]: Could we include the valid values here?
```suggestion
* `version` - (Optional) The version (revision) of the flow log. POssible values are `1` and `2`.
```

**Comment by aqche** [comment]
@katbyte thanks for the review, applied your suggestions!

**Pull request merged**